### PR TITLE
new Favourites 

### PR DESCRIPTION
**Pull request by VigneshPalanisamy**

### CODE REVIEW

File: worshipsongs/worshipsongs/AddFavoriteViewController.swift [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        // Dispose of any resources that can be recreated.
+    }
+    
+    @IBAction func valueChanged(_ sender: Any) {
+       self.navigationItem.rightBarButtonItem?.isEnabled = !(favoriteName.text?.isEmpty)!
+    }
+    
+    @IBAction func createFav(_ sender: Any) {
+        favoriteName.resignFirstResponder()
+        var favoriteKey = "favorite"
```
Comment (madasamys) [inline worshipsongs/worshipsongs/AddFavoriteViewController.swift]: Define constant

File: worshipsongs/worshipsongs/AddFavoriteViewController.swift [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    @IBAction func cancel(_ sender: Any) {
+       _ = self.navigationController?.popViewController(animated: false)
+    }
+    
+    fileprivate func confirmAlertController(_ message: String) -> UIAlertController
+    {
+        let confirmationAlertController =
+            UIAlertController(title: songTitle,
+                              message: message, preferredStyle: UIAlertControllerStyle.alert)
+        confirmationAlertController.addAction(self.getCancelAction(title: "ok"))
```
Comment (madasamys) [inline worshipsongs/worshipsongs/AddFavoriteViewController.swift]: Define ```ok``` in localized file

Comment (VigneshPalanisamy) [reply worshipsongs/worshipsongs/AddFavoriteViewController.swift]: it was there in cancelAction

File: worshipsongs/worshipsongs/ArtistSongsTitleTableViewController.swift [hunk 3]
```diff
@@ -73,10 +74,8 @@ class ArtistSongsTitleTableViewController: UITableViewController, UISearchBarDel
         let pressingPoint = longPressGesture.location(in: self.tableView)
         let indexPath = self.tableView.indexPathForRow(at: pressingPoint)
         if indexPath != nil && longPressGesture.state == UIGestureRecognizerState.began {
-            let viewController = storyboard?.instantiateViewController(withIdentifier: "favorite") as? ManageFavoritesController
-            viewController?.song = filteredSongModel[(indexPath?.row)!]
-            viewController?.modalPresentationStyle = UIModalPresentationStyle.overFullScreen
-            self.present(viewController!, animated: false, completion: nil)
+            addToFav = filteredSongModel[(indexPath?.row)!]
+            performSegue(withIdentifier: "manageFav", sender: self)
```
Comment (madasamys) [inline worshipsongs/worshipsongs/ArtistSongsTitleTableViewController.swift]: Define constant for identifier

File: worshipsongs/worshipsongs/ArtistsTableViewController.swift [hunk 4]
```diff
@@ -25,12 +25,17 @@ class ArtistsTableViewController: UITableViewController   {
         super.viewDidLoad()
         songTabBarController = self.tabBarController as? SongsTabBarViewController
         self.tabBarItem.title = "artists".localized
-        tableView.contentInset = UIEdgeInsetsMake(0, 0, (self.tabBarController?.tabBar.frame.height)!, 0)
         //refresh control
         refresh = UIRefreshControl()
         refresh.attributedTitle = NSAttributedString(string: "Refresh")
         refresh.addTarget(self, action: #selector(ArtistsTableViewController.refresh(_:)), for:UIControlEvents.valueChanged)
```
Comment (madasamys) [inline worshipsongs/worshipsongs/ArtistsTableViewController.swift]: Is really needed refresh control?

Comment (VigneshPalanisamy) [reply worshipsongs/worshipsongs/ArtistsTableViewController.swift]: yes after search

**Pull request merged**